components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/IWAAuthenticationUtil.java

```diff
@@ -58,21 +59,29 @@
     private static transient GSSCredential localIWACredentials;
     private static transient KerberosPrincipal serverPrincipal;
 
-    // Shared Map to hold IWA GSS credentials for respective Kerberos servers
-    private static transient Map<String, GSSCredential> gssCredentialMap = new ConcurrentHashMap<>();
     private static Log log = LogFactory.getLog(IWAAuthenticationUtil.class);
 
 
     public static void initializeIWALocalAuthenticator() throws GSSException, PrivilegedActionException, LoginException {
         RealmService realmService = dataHolder.getRealmService();
+        RealmConfiguration realmConfiguration = realmService.getBootstrapRealmConfiguration();
 
-        String servicePrincipalName =
-                realmService.getBootstrapRealmConfiguration().getUserStoreProperty(IWAConstants.SPN_NAME);
-        String servicePrincipalPassword =
-                realmService.getBootstrapRealmConfiguration().getUserStoreProperty(IWAConstants.SPN_PASSWORD);
+        // TODO : read this config from a file in registry

```

Comment:
We decided to move without involving registry and get user store domains directly from UI. So this to do is a deprecated now